src/command/global/settings.ts

```diff
@@ -0,0 +1,163 @@
+    }
+    return Color.SECONDARY;
+}
+
+async function createRows(discordId: string) {
+    const row = new MessageActionRow()
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.ENEMY)
+                .setLabel("Enemy Movements")
+                .setEmoji("☠️")
+                .setStyle(await getStyle(discordId, Toggles.ENEMY))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.WAR)
+                .setLabel("War Timer")
+                .setEmoji("🔫")
+                .setStyle(await getStyle(discordId, Toggles.WAR))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.QUEUE)
+                .setLabel("Empty Queue")
+                .setEmoji("🧠")
+                .setStyle(await getStyle(discordId, Toggles.QUEUE))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.REIMB)
+                .setLabel("Reimb Ready")
+                .setEmoji("💰")
+                .setStyle(await getStyle(discordId, Toggles.REIMB))
+        );
+
+    const row2 = new MessageActionRow()
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.EVENT)
+                .setLabel("New Event")
+                .setEmoji("❗")
+                .setStyle(await getStyle(discordId, Toggles.EVENT))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.MAIL)
+                .setLabel("New Mail")
+                .setEmoji("📧")
+                .setStyle(await getStyle(discordId, Toggles.MAIL))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.PAUSED)
+                .setLabel("Pause Notifications")
+                .setEmoji("💤")
+                .setStyle(await getStyle(discordId, Toggles.PAUSED))
+        );
+
+    return [row, row2];
+}
+
+function getCurrentSetting(setting: string, settings: NotificationSettings) {
+    if (setting == Toggles.WAR) {
+        return settings.war_flag;
+    }
+    if (setting == Toggles.QUEUE) {
+        return settings.queue_flag;
+    }
+    if (setting == Toggles.REIMB) {
+        return settings.reimb_flag;
+    }
+    if (setting == Toggles.ENEMY) {
+        return settings.enemy_flag;
+    }
+    if (setting == Toggles.MAIL) {
+        return settings.mail_flag;
+    }
+    if (setting == Toggles.EVENT) {
+        return settings.event_flag;
+    }
+    if (setting == Toggles.PAUSED) {
+        return settings.paused_flag;
+    }
+}
+
+export const handler = async (interaction: CommandInteraction) => {
+    const settingsRepository = getCustomRepository(
+        NotificationSettingsRepository
+    );
+    const response = await settingsRepository.getUserByDiscordId(
+        interaction.user.id
+    );
+    if (response && !response.valid_key) {
```

Comment:
this part is weird. If response _doesn't_ exists it still shows the settings. Also, _response_ isn't really the most concise name for the return value >.>

Reply:
good catch. and stop bullying my variable names thnx